### PR TITLE
support compact range options

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**
@siddontang @hhkbp2 PTAL

### CODE REVIEW

File: src/rocksdb_options.rs [hunk 1]
```diff
@@ -262,6 +262,30 @@ impl WriteOptions {
...
+pub struct CompactOptions {
+    pub inner: *mut DBCompactOptions,
+}
+
+impl CompactOptions {
+    pub fn new() -> CompactOptions {
+        unsafe { CompactOptions { inner: crocksdb_ffi::crocksdb_compactoptions_create() } }
+    }
+
+    pub fn set_exclusive_manual_compaction(&mut self, v: bool) {
```
Comment (siddontang) [inline src/rocksdb_options.rs]: exclusive means the range is `(start, end)`?

**Comment by siddontang** [comment]
LGTM

**Comment by siddontang** [comment]
PTAL @lishihai9017

File: src/rocksdb_options.rs (hunk 1 above)
Comment (zhangjinpeng1987) [inline src/rocksdb_options.rs]: It is means when manual compaction is running, other background compaction threads will stop. We need set it false in our case.

**Comment by zhangjinpeng1987** [comment]
@hhkbp2 PTAL

**Comment by siddontang** [comment]
Please fix the conflict @zhangjinpeng1987

**Comment by lishihai9017** [comment]
LGTM

**Comment by hhkbp2** [comment]
LGTM

**Pull request merged**